### ISSUE TITLE
dispatch error while `scitype` on an `SentinelArray` which is used by CSV.jl

### ISSUE THREAD

**Issue by OkonSamuel** [alan-turing-institute/ScientificTypes.jl#104]
```julia
julia> using MLJScientificTypes, SentinelArrays

julia> b =SentinelArray([1,2,3,4])
4-element SentinelArray{Int64,1,Int64,Missing,Array{Int64,1}}:
 1
 2
 3
 4

julia> scitype(b)
Unknown

julia> scitype(b, MLJScientificTypes.ScientificTypes.convention())
Unknown

julia> scitype(b, MLJScientificTypes.ScientificTypes.convention(), Val(trait(b)))
AbstractArray{Union{Missing, Count},1}

```
I thinking this is a julia bug. Making a small tweak to the code resolves the issue but i can't seem to understand why this is an issue in the first place.
@ablaom , @tlienart thoughts??

**Comment by OkonSamuel** [comment]
This issue has been fixed in the recent julia 1.5.1 patch release.
<pre><font color="#4E9A06"><b>julia&gt; </b></font>using MLJScientificTypes, SentinelArrays

<font color="#4E9A06"><b>julia&gt; </b></font>b =SentinelArray([1,2,3,4])
4-element SentinelArray{Int64,1,Int64,Missing,Array{Int64,1}}:
 1
 2
 3
 4

<font color="#4E9A06"><b>julia&gt; </b></font>scitype(b)
AbstractArray{Union{Missing, Count},1}

<font color="#4E9A06"><b>julia&gt; </b></font>scitype(b, MLJScientificTypes.ScientificTypes.convention())
AbstractArray{Union{Missing, Count},1}

<font color="#4E9A06"><b>julia&gt; </b></font>scitype(b, MLJScientificTypes.ScientificTypes.convention(), Val(trait(b)))
AbstractArray{Union{Missing, Count},1}</pre>

**Comment by OkonSamuel** [comment]
closing this. Since it's been resolved
